### PR TITLE
mangohud: 0.6.1 → 0.6.3

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
Update to the latest version: https://github.com/flightlessmango/MangoHud/releases/tag/v0.6.3

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@ZerataX

### CODE REVIEW

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 127469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mangohud</li>
  </ul>
</details>

**Comment by ZerataX** [comment]
I'll check in a bit ^^
btw added a home-manager module
https://github.com/nix-community/home-manager/commit/fb50102daf904c4ced33374816f9ee2cfde24c01

File: pkgs/tools/graphics/mangohud/default.nix [hunk 1]
```diff
@@ -61,7 +77,7 @@ stdenv.mkDerivation rec {
         glxinfo
         gnugrep
         gnused
-        pciutils
```
Comment (kira-bruneau) [inline pkgs/tools/graphics/mangohud/default.nix:61]: pciutils is no longer needed: https://github.com/flightlessmango/MangoHud/commit/f8fa73daf2748c69d1cb79916edb9f515e016b99#diff-77743d25e8e941dd6c2a92d819b80cfc171108ef8f088cebdaad2b45ad39c928L612-L613

File: pkgs/tools/graphics/mangohud/default.nix [hunk 2]
```diff
@@ -61,7 +77,7 @@ stdenv.mkDerivation rec {
...
+        lsof
```
Comment (kira-bruneau) [inline pkgs/tools/graphics/mangohud/default.nix:80]: `lsof` is required now:
https://github.com/flightlessmango/MangoHud/commit/ddfcb01c30790674e7316cae14bc28c4150610a4#diff-3454b244a6938af5e3c9c0d5e12d77cbda297a34aaa70e7f0841dd6678a5e735R79

**Review by kira-bruneau — COMMENTED** [review]
Sorry for the force-pushes. I noticed some runtime dependencies have changed.

File: pkgs/tools/graphics/mangohud/default.nix [hunk 3]
```diff
@@ -22,30 +22,49 @@
...
+  # Derived from subprojects/imgui.wrap
+  imgui = rec {
+    version = "1.81";
+    src = fetchFromGitHub {
+      owner = "ocornut";
+      repo = "imgui";
+      rev = "v${version}";
+      hash = "sha256-rRkayXk3xz758v6vlMSaUu5fui6NR8Md3njhDB0gJ18=";
+    };
+    patch = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/mangohud/default.nix:41]: This should use fetchpatch.

Comment (kira-bruneau) [reply pkgs/tools/graphics/mangohud/default.nix:41]: It's a zip, not a patch file. It's just referred to as a patch in https://github.com/flightlessmango/MangoHud/blob/master/subprojects/imgui.wrap.

**Review by zeratax — APPROVED** [review]
took me a second to get wayland running, but yeah works with wayland yay!

I think my wattage isn't being displayed correctly ^^ but since mangohud doesnt complain I assume the issue is elsewhere.

Switched to nvidia, so just now noticing that gpu utilization isn't being displayed, see also
https://github.com/NixOS/nixpkgs/pull/95225#pullrequestreview-468429627
```
...
MANGOHUD: Failed to open 64bit libnvidia-ml.so.1: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
MANGOHUD: Failed to load NVML
MANGOHUD: Failed to open 64bit libXNVCtrl.so.0: libXNVCtrl.so.0: cannot open shared object file: No such file or directory
MANGOHUD: XNVCtrl loader failed to load
MANGOHUD: Uploading is disabled (permit_upload = 0)
MANGOHUD: No battery found
```

but I guess that's stuff for another PR ^^

mangohud

**Comment by kira-bruneau** [comment]
> Switched to nvidia, so just now noticing that gpu utilization isn't being displayed, see also

I think that should be fixed by using `addOpenGLRunpath` (see [nvtop](https://github.com/NixOS/nixpkgs/blob/2892b7f640fbc222654fcf2b02886c9ab3bb3e7b/pkgs/tools/system/nvtop/default.nix#L24)). I don't have a Nvidia card to test this with though, could you confirm if this works?

**Comment by kira-bruneau** [comment]
Oh, wait... I think `addOpenGLRunpath` only works on binary files and MangoHud is a shell script.

**Comment by kira-bruneau** [comment]
What about this?

```patch
diff --git a/pkgs/tools/graphics/mangohud/default.nix b/pkgs/tools/graphics/mangohud/default.nix
index 1015b4cd1bd..8db2135af58 100644
--- a/pkgs/tools/graphics/mangohud/default.nix
+++ b/pkgs/tools/graphics/mangohud/default.nix
@@ -16,6 +16,7 @@
 , libX11
 , mangohud32
 , vulkan-headers
+, addOpenGLRunpath
 , glslang
 , makeWrapper
 , meson
@@ -105,6 +106,7 @@ in stdenv.mkDerivation rec {
   ];
 
   nativeBuildInputs = [
+    addOpenGLRunpath
     glslang
     makeWrapper
     meson
@@ -130,9 +132,11 @@ in stdenv.mkDerivation rec {
       "$out/share/vulkan/implicit_layer.d/MangoHud.x86.json"
   '';
 
-  # Support overlaying Vulkan applications without requiring mangohud to be installed
+  # Support Nvidia cards by adding OpenGL path and support overlaying
+  # Vulkan applications without requiring mangohud to be installed
   postFixup = ''
     wrapProgram "$out/bin/mangohud" \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ addOpenGLRunpath.driverLink ]} \
       --prefix XDG_DATA_DIRS : "$out/share"
   '';
```

**Comment by ZerataX** [comment]
omg yeah that does work, awesome!
gpu even properly displays wattage as well ^^

**Review by zeratax — APPROVED** [review]

**Comment by musfay** [comment]
Result of `nixpkgs-review pr 127469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mangohud</li>
  </ul>
</details>

**Pull request merged**